Source/path.h

```diff
@@ -5,14 +5,26 @@
  */
 #pragma once
 
+#include <array>
 #include <functional>
 
 #include <SDL.h>
 
 #include "engine/direction.hpp"
+#include "engine/displacement.hpp"
 #include "engine/point.hpp"
 
 namespace devilution {
+/** Specifies a 3x3 search area (including the starting tile). */
+constexpr std::array<Displacement, 9> SearchOffsets3x3 {
+	{
+	    // clang-format off
+		    { -1, -1 }, { 0, -1 }, { 1, -1 },
+		    { -1,  0 }, { 0,  0 }, { 1,  0 },
+		    { -1,  1 }, { 0,  1 }, { 1,  1 },
+	    // clang-format on
+	}
+};

```

Comment:
Can't this be done using a 3x3 rectangle?

I understand using individual points for more "complex" shapes (like quest entry points) since we don't have a general purpose `Region` abstraction, but for these simpler areas, it feels to me there is no reason to go with individual displacements in place of a `Rectangle` instance that is applied at tile space.

Reply:
Probably, I'll see what it looks like at some point over the weekend.